materialize-bigquery/sqlgen.go

```diff
@@ -239,7 +239,7 @@ ON {{ range $ind, $key := $.Keys }}
 	l.{{$key.Identifier}} = r.c{{$ind}}
 {{- end}}
 {{- if $.Document }}
-WHEN MATCHED AND r.c{{ Add (len $.Columns) -1 }} IS NULL THEN
+WHEN MATCHED AND JSON_TYPE(PARSE_JSON(r.c{{ Add (len $.Columns) -1 }}))='null' THEN
```

Comment:
Is there any performance impact when using these extra functions? I remember a time when we tried to allow primary keys to be nullable, and we did a (completely different) kind of `null` checking in load query key equality that was very detrimental for performance.

Reply:
It seems like we are serializing a json string with the value of `"null"` to indicate deletions, so is there a more direct way to check for that here? Something more along the lines of `r.cN = 'null'`? Or does that not work for some reason?